src/builder.ts

```diff
@@ -599,51 +643,53 @@ export class SchemaBuilder {
     if (config.rootTyping) {
       this.rootTypings[config.name] = config.rootTyping;
     }
-    return this.finalize(
-      new GraphQLObjectType({
-        name: config.name,
-        interfaces: () => interfaces.map((i) => this.getInterface(i)),
-        description: config.description,
-        fields: () => {
-          const allFieldsMap: GraphQLFieldConfigMap<any, any> = {};
-          const allInterfaces = interfaces.map((i) => this.getInterface(i));
-          allInterfaces.forEach((i) => {
-            const interfaceFields = i.getFields();
-            // We need to take the interface fields and reconstruct them
-            // this actually simplifies things becuase if we've modified
-            // the field at all it needs to happen here.
-            Object.keys(interfaceFields).forEach((iFieldName) => {
-              const { isDeprecated, args, ...rest } = interfaceFields[
-                iFieldName
-              ];
-              allFieldsMap[iFieldName] = {
-                ...rest,
-                args: args.reduce(
-                  (result: GraphQLFieldConfigArgumentMap, a) => {
-                    const { name, ...argRest } = a;
-                    result[name] = argRest;
-                    return result;
-                  },
-                  {}
-                ),
+    const objectTypeConfig = {
+      name: config.name,
+      interfaces: () => interfaces.map((i) => this.getInterface(i)),
+      description: config.description,
+      fields: () => {
+        const allInterfaces = interfaces.map((i) => this.getInterface(i));
+        const interfaceConfigs = allInterfaces.map((i) => i.toConfig());
+        const interfaceFieldsMap: GraphQLFieldConfigMap<any, any> = {};
+        interfaceConfigs.forEach((i) => {
+          Object.keys(i.fields).forEach((iFieldName) => {
+            const { extensions, ...objectFieldConfig } = i.fields[
+              iFieldName
+            ] as GraphQLFieldConfig<any, any> & {
+              extensions: { nexus: NexusFieldExtension };
+            };
+            const localModifications = modifications[iFieldName];
+            const fieldExtensionConfig = {
+              ...extensions.nexus.fieldConfig,
+            } as NexusOutputFieldConfig<any, any>;
+            // TODO(tim): Not sure if all this is really even necessary, t.modify is probably the
```

Comment:
Nice idiom `TODO(<person>)`, will remember that one on distributed team projects.